### ISSUE TITLE
Bot that downloads photos from an (encrypted) group chat and stores them in a folder

### ISSUE THREAD

**Issue by arthurlutz** [Kegsay/Matrix-NEB#23]
I was wondering if this python project would be a good starting point to write a bot that sits in an encrypted group chat and for every photo / video that is posted there, stores them to disk (which behind would be synced using syncthing, shared by members using syncting and served over the web by a nextcloud instance). This is for a family usecase as an alternative to WhatsApp and I realised some members were loosing time saving all the photos posted in the group chat one by one. 

I have more skills in python than go so I don't know if I can ignore #21

**Comment by createchange** [comment]
I wrote a set of shell and Python script and just have it pulling pictures from the underlying volume. It scrapes the the previous month (cron job) and FTP's them over to my Nextcloud instance. 

I would say that having the bot as an abstraction on top of things will actually make it more difficult to accomplish what you want.

**Comment by arthurlutz** [comment]
@createchange good point. A script that connects and does an extraction could be better and easier for this sort of task. Can you share your scripts ?

**Comment by arthurlutz** [comment]
Looking at your account I  found https://github.com/createchange/matrix-synapse-media-backup-to-nextcloud which seems to be what you are describing. Thanks for sharing. 

From what I understand, it's different from what I describe in the following :+1: 

* I don't have FTP access to the matrix server I'm using (matrix.org)
* The room in question is where the media is shared is E2E encrypted so I need to extract images via matrix and it's encryption keys which makes it a bit more tricky

**Comment by mztiq** [comment]
@arthurlutz I know this is old but did you find a solution yet? I'm trying to backup Images of E2E encrypted and non-encrypted chats too and couldn't get the solution of @createchange to work.

**Comment by arthurlutz** [comment]
I succeeded doing an archive with https://github.com/russelldavies/matrix-archive/
